arrows/core/transform_detected_object_set.cxx

```diff
@@ -213,23 +213,24 @@ box_around_box3d( const kwiver::vital::camera_perspective_sptr camera,
     projected_points.row(i) = camera->project(box3d.row(i)).transpose();
   }
 
-  vector_4d out_box;
-  out_box << (vector_2d)projected_points.colwise().minCoeff(),
-             (vector_2d)projected_points.colwise().maxCoeff();
+  auto out_box = vital::bounding_box<double>(
+    (vector_2d)projected_points.colwise().minCoeff(),
+    (vector_2d)projected_points.colwise().maxCoeff());
 
   return out_box;
 }
 
 // ---------------------------------------------------------------------------
-vector_4d
+vital::bounding_box<double>
 transform_detected_object_set::
 view_to_view( const kwiver::vital::camera_perspective_sptr src_camera,
               const kwiver::vital::camera_perspective_sptr dest_camera,
-              const vector_4d bounds ) const
+              const vital::bounding_box<double> bbox ) const
```

Comment:
It's probably better for efficiency to pass `bbox` as  `vital::bounding_box<double> const& bbox`.  The const reference avoids making a deep copy when pass it in.  This is a small enough structure that it probably doesn't make much difference, but it's good practice for anything bigger than the built in basic types.  The other exception here is that we usually pass shared pointer by value (though there is some debate about whether we should) since they are intended to make a light-weight shallow copy.

Also, the `const` can go before or after the type name.  It is now KWIVER style to put the const after the type name.  See here https://github.com/Kitware/kwiver/blob/master/doc/code-style.rst#type-names-types

Reply:
Ah, I see.  I've made the suggested pass by reference change here and elsewhere where it seemed appropriate (including your other suggested locations).

Gotcha, I've made that change as well.